megatron/training.py

```diff
@@ -864,79 +891,79 @@ def training_log(loss_dict, total_loss_dict, learning_rate, iteration,
        (iteration % args.tensorboard_log_interval == 0):
         timers.write(timers_to_log, writer, iteration,
                      normalizer=total_iterations)
-    if writer and (iteration % args.tensorboard_log_interval == 0):
-        writer.add_scalar('steps-vs-samples/y=steps,x=samples', iteration, args.consumed_train_samples)
+    if writer.is_enabled() and (iteration % args.tensorboard_log_interval == 0):
+        writer.add_scalar_to_tb('steps-vs-samples/y=steps,x=samples', iteration, args.consumed_train_samples)
         writer.add_scalar('steps-vs-samples/y=samples,x=steps', args.consumed_train_samples, iteration)
```

Comment:
I am curious why this `writer.add_scalar` is not changed to `writer.add_scalar_to_tb`. How are you deciding which ones to change?

Reply:
Hi, At the beginning, I use add_scalar_to_tb as a whilelist to avoid descend step specified in wandb. For wandb,  step must be monotonically increasing in each call, otherwise the step value is ignored during your call to log(). In other words, my initial patch does not support custom x-axes which is enabled in tensorboard. With latest patch , https://github.com/microsoft/Megatron-DeepSpeed/pull/416/commits/41696b63552895e49e7477282b39244e7264b86f , I integrated this feature so eliminate all add_scalar_to_tb invocation.